pkg/testing/fetcher_artifact.go

```diff
@@ -78,16 +78,18 @@ func (f *artifactFetcher) Fetch(ctx context.Context, operatingSystem string, arc
 	path := fmt.Sprintf("elastic-agent-%s-%s", version, suffix)
 	downloadSrc := fmt.Sprintf("%s%s", uri, path)
 	return &artifactResult{
-		doer: f.doer,
-		src:  downloadSrc,
-		path: path,
+		doer:      f.doer,
+		src:       downloadSrc,
+		path:      path,
+		fetchHash: true,
 	}, nil
 }
 
 type artifactResult struct {
-	doer httpDoer
-	src  string
-	path string
+	doer      httpDoer
+	src       string
+	path      string
+	fetchHash bool
```

Comment:
AFAICT, `fetchHash` will always be `true`.  So do we even need this field?  In other words, should we always just fetch the hash? I don't think it adds too much additional cost or risk?

Reply:
I wasn't sure if we want to use fetchers to get other artifacts beyond elastic-agent and so I didn't know if all kind of artifacts always have the hash in the same format. It could be premature optimization here, as you said right now this is always true